src/codegen/mod.rs

```diff
@@ -772,8 +772,26 @@ impl CodeGenerator for Type {
                     return;
                 }
 
-                tokens.append_all(quote! {
-                    pub type #rust_name
+                {
+                    let alias_style = ctx.options().default_alias_style;
```

Comment:
I think I'd prefer doing something like:

```rust
let alias_style = ctx.options.default_alias_style;

tokens.append_all(match alias_style {
    AliasVariation::TypeAlias => ...,
    AliasVariation::NewType | AliasVariation::NewTypeDeref => {
        assert!(ctx.options().rust_features.repr_transparent, "...");
        quote! { ... }
    }
});
```

That way you only lookup the option once, and match exhaustively (so that the per-type choice is easy to implement, and new variations also are less footgunny).